back-office/international/translations/translations.md

```diff
@@ -0,0 +1,140 @@
+
+Once you click on the "Edit" button, you go to the translation interface.
+
+### Particular cases
+
+- When the user chooses "Front office translations", a list of all the installed themes except "classic" + a special item called "Core translations" is displayed as well.
+- When the user chooses "Installed module translations", a list of all the installed modules is displayed as well.
+- When the user chooses "Email translations", a list containing "Subject" and "Message body" is displayed as well.
+  - If the user chooses "Message body", a list of all the installed themes + "Core" is displayed as well.
+  - This type sends the user to the legacy translation interface instead of the one described below.
+
+### Translation interface
+
+This interface displays two columns. In the left, there's a tree of translation domains according to the translation type the user chose before. 
+
+The tree is composed by separating translation domain parts using dots. Eg:
+
+```
+Admin
+├── Actions
+├── AdvParameters
+│   ├── Feature
+│   ├── Help
+│   └── Notification
+└── Global
+```
+
+The tree above represents the following tranlsation domains:
+
+- Admin.Actions
+- Admin.AdvParameters.Feature
+- Admin.AdvParameters.Help
+- Admin.AdvParameters.Notification
+- Admin.Global
+
+Only the leaves of this tree are clickable. On the right, the list of  wordings belonging to the currently selected domain are shown, each followed by a text field containing its translation to the selected language.
+
+The user can edit translations by filling out the fields, and save them.
+
+On the top of the page, there's a search box. When the user searches for a wording, the list is filtered, meaning that wordings not matching the search expression are no longer shown on the list and domains containing no matching wording are grayed out.
+
+#### Translation sources & storage
+
+This section covers where wordings are sourced from and where translations are stored to according to the translation type chosen.
+
+##### Generalities
+
+Usually and unless specified otherwise:
+
+- Wordings are sourced from XLF files stored somewhere, one file per translation domain.
+- User-edited translations are stored in the database
+
+##### Back office translations
+
+Wordings are sourced from the Core's _default catalogue_, which is stored as physical XLF files (one per translation domain) in `app/Resources/translations/default`. This catalogue is to be filtered by keeping only the translation domains whose name starts with **"Admin"**.
+
+Translations are retrieved from two sources: 
+
+1. The core: XLF translation files for the chosen language, stored in `app/Resources/translations/{locale_code}`, one per domain.
+2. The database: if translations were customized previously through this interface.
+
+If the same wording is translated in both core files and database, the database translation is shown.
+
+##### Front office translations
+
+This translation type has two different subcases:
```

Comment:
Are we okay that the front office translations now gathers 1/ the core translations (all frontend wordings), 2/ the theme translations, and 3/ the native modules translations (all front wordings)?

Reply:
No

Core domains start with "Shop", whereas module domains _end_ with "Shop".

Therefore:

- If the user chooses "Core translations", then it only gets the ones from the core.
- If the user chooses a theme, then it gets all the wordings found in the theme's templates, including any core wording also found there.

This has one caveat: you can't redefine a core translation for a given theme if it's not used in one of the theme's template files.

Example: If a given core wording is displayed in front office, but the wording is physically located in a PHP core file and not within one of the theme's templates, then the wording won't show up for that theme. On the contrary, the wording will show up if it's found in one of the theme's template.

The alternative would be to display all the core wordings, but there's no way to tell if they're actually used in the theme or not.

Reply:
I quote the summary email sent after the meeting to discuss this feature:

> Autrefois traductions du thème, la seconde entrée s'intitulera désormais traductions du front-office pour regrouper l'ensemble des wordings présents en front, soit les traductions du _core_ (front-end), du _thème_ (ensemble des wordings utilisés dans le template du thème) et des _modules natifs_ (front). [...] À l'export, le comportement reste le même : exporter les traductions du front office reviendra à exporter l'intégralité des catalogues Shop du _core_, Shop des _modules natifs_ et du catalogue des wordings embarqués dans le _thème_.